kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/KotlinSymbolProcessing.kt

```diff
@@ -71,77 +354,239 @@ class KotlinSymbolProcessing(
             )
         }
 
-        val analysisAPISession = buildStandaloneAnalysisAPISession(withPsiDeclarationFromBinaryModuleProvider = true) {
-            CoreApplicationEnvironment.registerExtensionPoint(
-                project.extensionArea,
-                KtResolveExtensionProvider.EP_NAME.name,
-                KtResolveExtensionProvider::class.java
-            )
-            (project as MockProject).registerService(
-                KtLifetimeTokenProvider::class.java,
-                KtAlwaysAccessibleLifeTimeTokenProvider::class.java
+        val (analysisAPISession, kotlinCoreProjectEnvironment, modules) = createAASession(compilerConfiguration)
+
+        val kspCoreEnvironment = KSPCoreEnvironment(analysisAPISession.project as MockProject)
+
+        // TODO: deferred symbols: use PSIs; they don't change.
+        // TODO: error handling, onError()
+        // TODO: performance
+        val project = analysisAPISession.project
+        val psiManager = PsiManager.getInstance(project)
+        var finished = false
+        var initialized = false
+        lateinit var codeGenerator: CodeGeneratorImpl
+        lateinit var processors: List<SymbolProcessor>
+        lateinit var newKSFiles: List<KSFile>
+        lateinit var newFileNames: Set<String>
+        while (!finished) {
```

Comment:
are we falling back to the previous design for multiple round processing? Just want to know if this conflicts with the plan to do multiple round within Gradle runner.

Reply:
IIRC the plan is to do it here, in the unified entry point of production and test. There will be multilple callers in addition to gradle and we want the setup to be as easy as possible.